pkg/api/v1/types.go

```diff
@@ -589,28 +597,23 @@ func ConvertInternalUserToExternal(internalUser *kubermaticv1.User, includeSetti
 	}
 
 	for _, binding := range userBindings {
-		bindingAlreadyExists := false
-		for _, pg := range apiUser.Projects {
-			if pg.ID == binding.Spec.ProjectID && pg.GroupPrefix == binding.Spec.Group {
-				bindingAlreadyExists = true
-				break
-			}
-		}
-		if !bindingAlreadyExists {
-			groupPrefix := ExtractGroupPrefix(binding.Spec.Group)
-			apiUser.Projects = append(apiUser.Projects, ProjectGroup{ID: binding.Spec.ProjectID, GroupPrefix: groupPrefix})
-		}
+		groupPrefix := ExtractGroupPrefix(binding.Spec.Group)
+		pg := ProjectGroup{ID: binding.Spec.ProjectID, GroupPrefix: groupPrefix}
+		apiUser.Projects = append(apiUser.Projects, pg)
 	}
 
 	for _, binding := range groupBindings {
-		bindingAlreadyExists := false
-		for _, p := range apiUser.Projects {
-			if p.ID == binding.Spec.ProjectID {
-				bindingAlreadyExists = true
+		overwriteRole := false
+		for _, pg := range apiUser.Projects {
+			if pg.ID == binding.Spec.ProjectID {
+				// Check if role from group binding is more permissive.
+				if RolePriority[binding.Spec.Role] > RolePriority[pg.GroupPrefix] {
+					overwriteRole = true
+				}
 				break
 			}
 		}
-		if !bindingAlreadyExists {
+		if overwriteRole {
```

Comment:
I think now you are adding projects only if the role priority is higher. If a userprojectbinding does not exist, you are not adding a project through GPBs

Reply:
Oh man, u r right. Let me fix this.